### ISSUE TITLE
Importing from bank when a bag is full stops progress [BUG] 

### ISSUE THREAD

**Issue by rbpaiett** [niketa-wow/farming-bar#8]
**Describe the bug**
When using the banking shortcuts to import items from the bank to your inventory, if a bag gets full, the import halts and you have to click again to import the remainder, even if there's room in another bag.

**Client**
113.4 (Beta version of addon).

**To Reproduce**
1. Use shortcuts to import from bank to inv.
2. One of your bags gets full.
3. Import process stops.
4. Use shortcuts again to complete import.

**Expected behavior**
Use banking shortcuts with interruption even when only one bag is full but there's room in other bags.

**Screenshots**
Can be provided if needed.

**Additional context**
N/A

**Comment by niketa-wow** [comment]
1. When this happens, how many items are left to withdraw? A partial stack? A full stack? Or more than a full stack?

2. Are there any errors? Lua errors? Add-on error message? Does the moving frame show complete and hide itself or stay stuck on the screen until you try again?

3. Which bag is being filled? The backpack or one of the other bag slots?

**Comment by niketa-wow** [comment]
I'm unable to reproduce this error in neither retail nor classic. I'll need more information to troubleshoot the problem.

If you have a screen capture and can make a short video, that would help. Otherwise, screenshots and some more details about the conditions before and after the process.

**Comment by rbpaiett** [comment]
I'll keep watch.  At worst, it's a mild inconvenience that's recovered by simply clicking again and it loads the remainder from the bank.  I'll look for scenarios where it happens all the time.  Otherwise as I said, simply a mild inconvenience.

**Comment by rbpaiett** [comment]
Ok, got it to repeat:

1.  Make a shopping list that needs 3 items.
2.  Make sure there are only two slots available in the backpack.
3.  Put all items in the bank.
4.  Use the banking shortcut to pull the needed items into your inventory.
5.  The tool will pull the first two items and put them in the backpack.  The process will stop because there's no more room in the backpack.  Use the shortcut again and the remaining item will be put in the next available bag in order from the bag next to the backpack in the toolbar and so forth.

I've also tested it where the backpack is full and only two slots are available in the next bag in order for a 3 item shopping list.  

Basically in this scenario, there has to be 3 slots available in one bag to pull a 3 item shopping list from the bank.

In the case of just a single item slot (like for instance 40 wool cloth), if you have only one slot available in the backpack, it will fill the one slot with 20 wool cloth, then put the remaining 20 in the next available bag without a prompt for a bag being full.

**Comment by niketa-wow** [comment]
I tried replicating the issue again, and it's very sporadic for me. Sometimes it works just fine, others I'm running across the issue with the full bag. Since I can't reproduce it consistently, I'm having a hard time finding out what the real issue is.

My best guess, however, is that when it's checking how many free slots are in the bag, it's happening too quickly while a previous item is about to be deposited there.  For example, if you have 1 slot left in a bag, then you try to fill those with two items, the first time around the first item is going to see there's a free slot and then move it, but before the move completes the second item is checking if there's a free slot in the bag and it sees the same free slot. Then when the second one moves, the first beats it to the punch and you get the error.

Because of this, the best thing I can think of to fix it is adding a delay after each move in shopping lists, the same way there is one with mixed items. So, this *might* fix the issue, but it'll be a little slower moving shopping lists from your bank to your bags. It should be about the same as moving the mixed items though, since that already had a delay (which was needed to make sure the correct amount is withdrawn).

Anyway, I'm going to attach a version for you to try out here with this change. Since it's the only thing I've changed, I don't want to push it even as a beta release until I'm sure this is even making a difference. If everything is good for you after a little bit, I'll release it to everyone or incorporate it for the final release.

This zip should be packaged for classic.
[FarmingBar.zip](https://github.com/niketa-wow/farming-bar/files/4691251/FarmingBar.zip)

**Comment by niketa-wow** [comment]
Also I just wanted to add the speed of depositing items from bags to bank hasn't changed since that's a lot easier to deal with, it usually doesn't require any delays.

**Comment by rbpaiett** [comment]
The version you sent me works like a charm, thanks!

**Comment by niketa-wow** [comment]
Awesome. Closing this issue. If it pops up again, just let me know. Thank you for your help in identifying bugs.
